### PR TITLE
fix include in settings.h

### PR DESCRIPTION
**Pull request by mszczodrak**

### CODE REVIEW

**Comment by mszczodrak** [comment]
@nibanks can you please suggest how to address the Windows build failure?

**Comment by nibanks** [comment]
It looks like you have some casting issues? Maybe you need to rebase to the latest? Can't really tell right now, since I'm on my phone. I can take a detailed look in and hour or so.

**Comment by mszczodrak** [comment]
Thanks @nibanks. It's a strange error since the change only fixes the include line.
Is there a way I can reproduce this setup at Linux or Mac?

**Comment by nibanks** [comment]
I don't know of any way to reproduce on Linux or Mac. Do none of you guys have a Windows machine?

**Comment by nibanks** [comment]
So, I don't know why it is complaining now, but I am able to repro on my Windows Driver PR after I sync to the latest and fix this header include as well. The issue is that the `enum` isn't automatically converted to a `uint16_t` in [meshcop_dataset.cpp](https://github.com/mszczodrak/openthread/blob/3c0e601b94df2e260f6681c434e0ea7b4e831d5f/src/core/thread/meshcop_dataset.cpp).

You need to wrap each `enum` in a `static_cast`, like this:

```
ThreadError Dataset::Restore(void)
{
    return otPlatSettingsGet(mInstance, static_cast<uint16_t>(mType == Tlv::kActiveTimestamp ? kKeyActiveDataset : kKeyPendingDataset), 0, mTlvs,
                             &mLength);
}
```

File: include/platform/settings.h [hunk 1]
```diff
@@ -36,7 +36,7 @@
 #define OT_PLATFORM_SETTINGS_H 1
 
 #include <stdint.h>
```
Comment (nibanks) [inline include/platform/settings.h]: Can we remove the `stdint.h` include, since `openthread-types.h` already includes it?

**Comment by nibanks** [comment]
I had to make these changes in [my most recent commit](https://github.com/openthread/openthread/pull/817/commits/cf36a57cde2840d68cdf47f92c46011ca6cb6b86) for the Windows Driver PR in order for it to compile.

**Pull request closed without merging**